CorePayments/src/main/java/com/paypal/android/corepayments/analytics/AnalyticsService.kt

```diff
@@ -1,34 +1,64 @@
 package com.paypal.android.corepayments.analytics
 
+import android.content.Context
 import android.util.Log
+import com.paypal.android.corepayments.CoreConfig
 import com.paypal.android.corepayments.Environment
-import com.paypal.android.corepayments.Http
-import com.paypal.android.corepayments.HttpRequestFactory
-import java.util.*
+import com.paypal.android.corepayments.PayPalSDKError
+import com.paypal.android.corepayments.SecureTokenServiceAPI
+import com.paypal.android.corepayments.TrackingEventsAPI
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import java.util.UUID
 
-internal class AnalyticsService(
+class AnalyticsService internal constructor(
     private val deviceInspector: DeviceInspector,
     private val environment: Environment,
-    private val http: Http,
-    private val httpRequestFactory: HttpRequestFactory
+    private val trackingEventsAPI: TrackingEventsAPI,
+    private val secureTokenServiceAPI: SecureTokenServiceAPI,
+    private val scope: CoroutineScope
 ) {
 
-    internal suspend fun sendAnalyticsEvent(name: String, clientID: String) {
-        val timestamp = System.currentTimeMillis()
+    constructor(
+        context: Context,
+        coreConfig: CoreConfig,
+        dispatcher: CoroutineDispatcher = Dispatchers.IO
```

Comment:
Does this need to be in the public constructor if no feature clients are setting it?

Reply:
I'll double check this. It got weird with testing lol.

Reply:
Yeah I was able to update and make another convenience constructor to discourage changing the coroutine dispatcher.

I do have a feeling this class will change a lot though, and it won't use coroutines once we migrate to `WorkManager` and let it do the heavy lifting with thread scheduling. All the `suspend` method invocations will move into a Work Manager task.